prototypes/ScheduledMergesTestQLS.hs

```diff
@@ -84,6 +81,48 @@ test_regression_empty_run =
         -- finish merge
         LSM.supply lsm 16
 
+-- | Covers the case where a run ends up too small for a level, so it gets
+-- merged again with the next incoming runs.
+-- That merge gets completed by supplying credits.
+test_merge_again_with_incoming :: IO ()
+test_merge_again_with_incoming =
+    runWithTracer $ \tracer -> do
+      stToIO $ do
+        lsm <- LSM.new
+        let ins k = LSM.insert tracer lsm k 0
+        -- get something to 3rd level (so 2nd level is not levelling)
+        -- (needs 5 runs to go to level 2 so the resulting run becomes too big)
+        traverse_ ins [101..100+(5*16)]
+        -- get a very small run (4 elements) to 2nd level
+        replicateM_ 4 $
+          traverse_ ins [201..200+4]
+        -- get another run to 2nd level, which the small run can be merged with
+        traverse_ ins [301..300+16]
+        -- complete the merge
+        LSM.supply lsm 32
```

Comment:
Should it not complete in 16 steps?

Reply:
I think you're right, I just supplied more as the batch size is 32 (and supplying too much doesn't hurt). But 16 should work, independent of the batch size. :+1: